src/mockServiceWorker.js

```diff
@@ -74,11 +74,22 @@ self.addEventListener('message', async function (event) {
   }
 })
 
-self.addEventListener('fetch', async function (event) {
+self.addEventListener('fetch', function (event) {
   const { clientId, request } = event
   const requestClone = request.clone()
   const getOriginalResponse = () => fetch(requestClone)
 
+  // Bypass navigation requests.
+  if (request.mode === 'navigate') {
```

Comment:
Navigation requests can be bypassed early, so I'm moving them outside of the `event.respondWith` promise. This way navigation requests are performed as-is, not within the worker's scope.